File: SortingAlgorithms/BubbleSort/bubble.py
import random as rand
import os

#forms the number array from file
filePath = os.path.dirname(os.getcwd())+"/testinput.txt"
inputData = open(filePath,'r').read()
nums = inputData.strip().split()
nums = [int(num) for num in nums]

#sorts using bubble sort
numSwaps = -1
while numSwaps!=0:
    numSwaps=0
    for i in range(len(nums)-1):
        if nums[i] > nums[i+1]:
            nums[i],nums[i+1] = nums[i+1],nums[i]
            numSwaps+=1
        
        
print(nums)

File: SortingAlgorithms/BubbleSort/bubble.cpp
#include <fstream>
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int main(){
    ifstream unsortedData ("../testinput.txt");
    vector<int> nums;
    
    //populate vector with input data
    int temp;
    while(unsortedData>>temp) nums.push_back(temp);
    unsortedData.close();

    //sort vector
    int numSwaps = -1;
    while(numSwaps){
        numSwaps=0;
        for(int i = 0; i<nums.size()-1;++i){
            if(nums[i] > nums[i+1]){
                swap(nums[i],nums[i+1]);
                ++numSwaps;
            }
        }
    }

    //print vector
    for(int num:nums)
        cout << num << " ";
    cout<<"\n";
}